tap_intercom/stream.py

```diff
@@ -21,40 +21,63 @@ def do_sync(self, tap_stream_id: str, state: Optional[datetime] = None):
             state=state, tap_stream_id=tap_stream_id
         )
         with singer.metrics.record_counter(tap_stream_id) as counter:
+            # The Intercom API doesn't allow for iterating through more than 10k companies
+            # using the list endpoint with pagination, instead requiring us to use their
+            # special "scroll" endpoint.
+            if tap_stream_id == "companies":
```

Comment:
I like `max(latest_updated_record, replication_value)` approach very much. It makes the code easier to understand. But we also handled the unordered data on line 62-67. if you can merge the two parts (companies and other endpoint). It would be great.